src/Core/Algorithms/ParticleInCell/GravitySimulationAlgo.cc

```diff
@@ -0,0 +1,404 @@
+/*
+        out_x.open ("/home/kj/Simulation-output/pos_x.txt");
+        out_y.open ("/home/kj/Simulation-output/pos_y.txt");
+        out_z.open ("/home/kj/Simulation-output/pos_z.txt");
+*/
+/*
+
+        for(int i=0; i<num_particles; i++)
+            {
+            out_x << pos_x[i] <<"\n";
+            out_y << pos_y[i] <<"\n";
+            out_z << pos_z[i] <<"\n";
+            }
+*/
+        }
+
+    for(j=1; j<iterations; j++)                       //main loop
+        {
+        iterate_end = thread_count;
+                                                      //Create (encode) acc_field
+        Generate_acc_field(pos_x, pos_y, pos_z, vel_x, vel_y, vel_z);
+
+                                                      //inner loop
+        for(int i=0; i<thread_count; i++) t[i] = thread(move_particles, i);
+        for(int i=0; i<thread_count; i++) t[i].join();
+                                                      //Inner loop end
+
+        for(int i=0; i<thread_count; i++)             //If all particles are dead, end the main loop
+            {
+            iterate_end -= t_alive[i];
+            t_alive[i] = 0;
+            }
+
+        if(iterate_end == thread_count) j = iterations;
+        else                                          //If at least one particle is still alive, continue processing the main loop
+            {
+                                                      //Set up the output matrices to contain the xyz position buffers
+            if(j==buffer_size)
+                {
+                double *data0 = output_mat_0_ref->data();
+                double *data1 = output_mat_1_ref->data();
+                double *data2 = output_mat_2_ref->data();
+                data0 = (double*)buffer_pos_x;
+                data1 = (double*)buffer_pos_y;
+                data2 = (double*)buffer_pos_z;
+                }
+          
+//            if(save)
+            if(save && j==buffer_size)
+                {
+                for(int i=0; i<num_particles; i++)
+                    {
+/*
+
+                    out_x << pos_x[i] <<"\n";
+                    out_y << pos_y[i] <<"\n";
+                    out_z << pos_z[i] <<"\n";
+*/
+                    out_x << buffer_pos_x[i] <<"\n";
+                    out_y << buffer_pos_y[i] <<"\n";
+                    out_z << buffer_pos_z[i] <<"\n";
+                    }
+                }
+            }
+        }                                             //main loop end
+
+    if(save)
+        {
+        out_x.close();
+        out_y.close();
+        out_z.close();
+        }
+
+    auto end_wall = chrono::high_resolution_clock::now();
+    chrono::duration<double> time_taken = end_wall - start_wall;
+    printf("Program took %.6f seconds Wall Clock time\n", time_taken.count());
+
+    clock_t end = clock();
+    printf("Program took %.3f seconds CPU time\n",(double)((end+0.0-start)/CLOCKS_PER_SEC));
+
+    output[Variables::OutputMatrix]=output_mat_0;
```

Comment:
Variable name (OutputMatrix) must be unique for each output, and must match the output port name

Reply:
`output[XCoordinates] = output_mat_0;`
etc.